lua/crates/completion/common.lua

```diff
@@ -126,6 +126,73 @@ local function complete_features(crate, cf, versions)
     }
 end
 
+---@param prefix string
+---@param span Span
+---@param line integer
+---@param kind WorkingCrateKind?
+---@return CompletionList?
+local function complete_crates(prefix, span, line, kind)
+    if #prefix < state.cfg.completion.crates.min_chars then
+        return
+    end
+
+    ::poll::
```

Comment:
I would prefer to use a `while` loop instead of `goto`. The only other places `goto` statements are used are as a replacement for `continue` statements.